components/common/carousel/carousel.tsx

```diff
@@ -25,30 +25,30 @@ const Carousel: React.FC<Props> = ({
 
     const nextSlide = () => {
         setPrevButtonDisabled(false);
-        if (slideIndex !== dataSlides.length) {
-            setSlideIndex(slideIndex + 1);
-        } else if (slideIndex === dataSlides.length) {
-            if (infiniteSlide) {
-                setSlideIndex(1);
+
+        const newSlideIndex = slideIndex === dataSlides.length && infiniteSlide ? 1 : (slideIndex !== dataSlides.length && slideIndex + 1) || undefined;
+
+        if (newSlideIndex) {
+            setSlideIndex(newSlideIndex);
+
+            if (onNextSlideClick) {
+                onNextSlideClick(newSlideIndex);
             }
         }
-        if (onNextSlideClick) {
-            onNextSlideClick();
-        }
     };
 
     const prevSlide = () => {
         setNextButtonDisabled(false);
-        if (slideIndex !== 1) {
-            setSlideIndex(slideIndex - 1);
-        } else if (slideIndex === 1) {
-            if (infiniteSlide) {
-                setSlideIndex(dataSlides.length);
+
+        const newSlideIndex = slideIndex === 1 && infiniteSlide ? dataSlides.length : (slideIndex !== 1 && slideIndex - 1) || undefined;
```

Comment:
nit: this kind of tertiary conditionals are hard to read, what if you self-document the code by storing in a variable with a meaningful name each of the options, like `const x = theCondition ? meaningfulName1 : meaningfulName2;`

Reply:
Updated!

Reply:
excelent, thanks!